kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/VirtualKafkaClusterReconciler.java

```diff
@@ -574,4 +572,16 @@ private VirtualKafkaCluster checkTlsConfigConsistency(Context<VirtualKafkaCluste
         return null;
     }
 
+    private static ClusterIP.Protocol getIngressProtocol(KafkaProxyIngress proxyIngress) {
+        ClusterIP clusterIP = proxyIngress.getSpec().getClusterIP();
+        if (clusterIP != null) {
+            return clusterIP.getProtocol();
+        }
+        LoadBalancer loadBalancer = proxyIngress.getSpec().getLoadBalancer();
+        if (loadBalancer != null) {
+            return ClusterIP.Protocol.TLS;
```

Comment:
Keith's improvement getting close to merging so we can use a custom enum that isn't tied to ClusterIP https://github.com/fabric8io/kubernetes-client/pull/7046 :+1: :tada: